Source/JavaScriptCore/builtins/JSIteratorPrototype.js

```diff
@@ -94,15 +100,21 @@ function take(limit)
     if (!@isObject(this))
         @throwTypeError("Iterator.prototype.take requires that |this| be an Object.");
 
-    var numLimit = @toNumber(limit);
-    if (numLimit !== numLimit)
+    var iterated = this;
```

Comment:
Let's just use `this` directly in this function instead of `iterated` when possible.